src/main/java/com/training/web/pages/flipkart/ResultPage.java

```diff
@@ -59,4 +62,36 @@ public ResultPage clickNextPage() {
         Collections.sort(sortedPrice);
         return sortedPrice;
     }
+
+    public List<WebElement> getProductsList() {
+        wait.until(ExpectedConditions.elementToBeClickable(nextPageButton));
+        wait.until(ExpectedConditions.elementToBeClickable(productResults));
+        List<WebElement> productList=driver.findElements(productResults);
+        return productList;
+    }
+
+    public ProductPage clickProduct(int itemNumber, List<WebElement> productList) {
```

Comment:
always the list should be first parameter followed by the index. That would make it more readable